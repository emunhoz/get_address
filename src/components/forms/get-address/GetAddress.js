import React from 'react'
import PropTypes from 'prop-types'
import Input from '../../input'
import Button from '../../button'
import { useForm } from 'react-hook-form'
import { RHFInput } from 'react-hook-form-input'
import * as S from './styled'

export const GetAddress = ({ handleSubmitForm, loading }) => {
  const { register, handleSubmit, setValue, errors } = useForm({
    mode: 'onChange'
  })

  return (
    <S.Form onSubmit={handleSubmit(handleSubmitForm)}>
      <S.InputWrapper>
        <RHFInput
          as={<Input id='cep' name='cep' label='CEP' />}
          name='cep'
          mask='11111-111'
          className={errors.cep ? 'error' : null}
          placeholder='Digite seu CEP'
          setValue={setValue}
          register={register({ required: true, pattern: /[0-9]{5}-[\d]{3}/g })}
        />
        {errors.cep && <S.HelpMessage>Campo necess√°rio</S.HelpMessage>}
      </S.InputWrapper>
      <Button>{loading ? <S.Spinner /> : 'Buscar'}</Button>
    </S.Form>
  )
}

GetAddress.propTypes = {
  handleSubmitForm: PropTypes.func,
  loading: PropTypes.bool
}
