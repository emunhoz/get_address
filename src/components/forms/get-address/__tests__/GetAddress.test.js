import React from 'react'
import { fireEvent, render, wait } from '@testing-library/react'

import { GetAddress } from '../GetAddress'

describe('Component: GetAddress', () => {
  let wrapper

  beforeEach(() => {
    wrapper = render(<GetAddress />)
  })

  it('render the component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('show needed help message', async () => {
    const { getByText } = wrapper
    const button = getByText('Buscar')

    fireEvent.submit(button)

    await wait(() => {
      expect(getByText('Campo necess√°rio')).toBeInTheDocument()
    })
  })
})
