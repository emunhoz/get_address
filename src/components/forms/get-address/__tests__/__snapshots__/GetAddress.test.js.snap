// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: GetAddress render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-AykKE diVPYj"
      >
        <div
          class="sc-AykKF boDatO"
        >
          <input
            class="sc-AykKC kbTzwD"
            data-testid="input-search"
            id="cep"
            mask="11111-111"
            maxlength="9"
            name="cep"
            placeholder="Digite seu CEP"
            size="9"
            value=""
          />
        </div>
        <button
          class="sc-AykKD kFlWLJ"
        >
          Buscar
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-AykKE diVPYj"
    >
      <div
        class="sc-AykKF boDatO"
      >
        <input
          class="sc-AykKC kbTzwD"
          data-testid="input-search"
          id="cep"
          mask="11111-111"
          maxlength="9"
          name="cep"
          placeholder="Digite seu CEP"
          size="9"
          value=""
        />
      </div>
      <button
        class="sc-AykKD kFlWLJ"
      >
        Buscar
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
